src/optimizer/analyze.cpp

```diff
@@ -0,0 +1,144 @@
+#include <algorithm>
+#include <cstring>
+#include <functional>
+#include <limits>
+#include <memory>
+#include <random>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "optimizer/analyze.h"
+
+namespace terrier {
+
+template <typename T>
+terrier::optimizer::ColumnStats CalculateColumnStats(catalog::db_oid_t database_id, catalog::table_oid_t table_id,
+                                                     catalog::col_oid_t columnID, storage::SqlTable *sql_table,
+                                                     terrier::transaction::TransactionContext *scan_txn,
+                                                     unsigned mostFrequentNumber) {
+  // Stats collecting temporary data-structures
+  std::unordered_map<T, unsigned> counting_map;
```

Comment:
Why are we using this instead of [`terrier::optimizer::Histogram`](https://github.com/cmu-db/terrier/blob/master/src/include/optimizer/statistics/histogram.h)?